test/jellyfish_web/controllers/component_controller_test.exs

```diff
@@ -297,4 +370,22 @@ defmodule JellyfishWeb.ComponentControllerTest do
   defp assert_no_hls_path(room_id) do
     assert {:error, :room_not_found} = HLS.EtsHelper.get_hls_folder_path(room_id)
   end
+
+  defp add_files_for_s3_upload(room_id) do
+    {:ok, hls_dir} = HLS.EtsHelper.get_hls_folder_path(room_id)
+    assert :ok = wait_for_folder(hls_dir, 1000)
+
+    for filename <- @files, do: :ok = hls_dir |> Path.join(filename) |> File.write(@body)
+  end
+
+  defp wait_for_folder(_hls_dir, milliseconds) when milliseconds < 0, do: {:error, :timeout}
+
+  defp wait_for_folder(hls_dir, milliseconds) do
+    if File.exists?(hls_dir) do
+      :ok
+    else
+      Process.sleep(100)
+      wait_for_folder(hls_dir, milliseconds - 100)
+    end
+  end
```

Comment:
Do we really need this? Couldn't we rely on some notifications etc?

Reply:
Element responsible for creating this folder is HLS endpoint so this is a tough one. I will look at it again.

Reply:
It's impossible right now because file creation has to happen before engine termination